tephra-core/src/main/java/org/apache/tephra/TransactionContext.java

```diff
@@ -97,6 +127,7 @@ public void start() throws TransactionFailureException {
                                        txAware.getTransactionAwareName(), currentTx.getTransactionId());
         LOG.warn(message, e);
         txClient.abort(currentTx);
+        currentTx = null;
```

Comment:
this is not strictly part of this new feature. Just fixing this along the way. Normally, after we abort the tx, we set currentTx = null. I thought it would be best to do that here, too